File: lista-1-python-go/exercicio-7/7-socket-client.go
// socket-client project 1-socket-client.go
package main

import (
	"fmt"
	"net"
)

const (
	SERVER_HOST = "localhost"
	SERVER_PORT = "12345"
	SERVER_TYPE = "tcp"
)

func server_send(connection net.Conn, err error, message string) {
	_, err = connection.Write([]byte(message))
	buffer := make([]byte, 1024)
	mLen, err := connection.Read(buffer)
	if err != nil {
		fmt.Println("Error reading:", err.Error())
	}
	fmt.Println("Received: ", string(buffer[:mLen]))
}

func server_receive(connection net.Conn, err error) {
	buffer := make([]byte, 1024)
	mLen, err := connection.Read(buffer)
	if err != nil {
		fmt.Println("Error reading:", err.Error())
	}
	fmt.Println("Received: ", string(buffer[:mLen]))
}

func main() {
	var name string
	var age string
	var service_time string

	// establish connection
	connection, err := net.Dial(SERVER_TYPE, SERVER_HOST+":"+SERVER_PORT)

	if err != nil {
		panic(err)
	}

	fmt.Println("Enter the employee name: ")
	fmt.Scanln(&name)
	fmt.Println("Enter the age: ")
	fmt.Scanln(&age)
	fmt.Println("Enter the service time: $")
	fmt.Scanln(&service_time)

	server_send(connection, err, name)
	server_send(connection, err, age)
	server_send(connection, err, service_time)
	server_receive(connection, err)

	defer connection.Close()
}

File: lista-1-concorrente/exercicio-7/7-socket-client.py
import socket

SERVER = "127.0.0.1"
PORT = 12345

client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
client.connect((SERVER, PORT))

while True:
    name = input("Enter your name: ")
    client.send(name.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    age = input("Enter your age: ")
    client.send(age.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    service_time = input("Enter your service time: ")
    client.send(service_time.encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    client.send("CONFIRM".encode())
    print("SERVER RESPONSE: " + client.recv(1024).decode())

    break
client.close()
